backstop_data/engine_scripts/puppet/page-objects/crm-page.js

```diff
@@ -104,6 +116,25 @@ module.exports = class CrmPage {
       this.engine.click(selector),
       this.engine.waitForNavigation()
     ]);
+    await this.waitForWYSIWYG();
+    await this.waitForDatePicker();
+    await this.fixLayoutForTables();
```

Comment:
why are we doing this all this stuff in `clickAndWaitForNavigation`? This method is now doing way much more than what it was supposed to do. Why can't we explicitly call these methods individually, in the scripts, on a case-by-case basis?

Reply:
This is a valid point. `clickAndWaitForNavigation` is doing what is expected from him. But if we remove these functions from the function it will be very hard to find the scenarios where we need them explicitly. Also for each function( `waitForWYSIWYG, waitForDatePicker, fixLayoutForTables` the actual computation only works inside a valid if condition. (i.e. If the page have elements to wait for the event then only they will wait). Since there are more than 300 cases I feel this is a fail safe way!.
We can however change the name of the function which make more sense. Something like `clickAndWaitForPageLoad()` will be a better option. What do you think?